### PR TITLE
Sd/cleanup

### LINKED ISSUE #82: Pkg.test() on windows with Julia-0.5 fails
```
julia> Pkg.test("FileIO")
INFO: Computing test dependencies for FileIO...
INFO: Installing FactCheck v0.4.3
INFO: Testing FileIO
FileIO
  > OS
  > DataFormat
  > streams
  > query
  > multiple libs
  > STL detection
  > PLY detection
  > Multiple Magic bytes
  > AVI Detection
  > RDA detection
    Failure :: (line:272) :: RDA detection :: fact was false
      Expression: FileIO.detect_rdata(io) --> true
        Expected: true
        Occurred: false
  > Format with function for magic bytes
  > Load
  > Save
  > Ambiguous extension
  > Absent file
  > Not installed
Library NotInstalled is not installed but can load format: FileIO.File{FileIO.Da
taFormat{:NotInstalled}}("test.not_installed")
should we install NotInstalled for you? (y/n):
Library NotInstalled is not installed but can load format: FileIO.File{FileIO.Da
taFormat{:NotInstalled}}("test.not_installed")
should we install NotInstalled for you? (y/n):
invalid is not a valid choice. Try typing y or n
should we install NotInstalled for you? (y/n):
Out of 99 total facts:
  Verified: 98
  Failed:   1
ERROR: LoadError: FactCheck finished with 1 non-successful tests.
 in exitstatus() at C:\Users\julia\AppData\Local\JuliaPro-0.5.0.1\pkgs-0.5.0.1\v
0.5\FactCheck\src\FactCheck.jl:568
 in include_from_node1(::String) at .\loading.jl:488
 in process_options(::Base.JLOptions) at .\client.jl:262
 in _start() at .\client.jl:318
while loading C:\Users\julia\AppData\Local\JuliaPro-0.5.0.1\pkgs-0.5.0.1\v0.5\Fi
leIO\test\runtests.jl, in expression starting on line 10
===============================[ ERROR: FileIO ]================================


failed process: Process(`'C:\Users\julia\AppData\Local\JuliaPro-0.5.0.1\Julia-0.
5.0\bin\julia' -Cx86-64 '-JC:\Users\julia\AppData\Local\JuliaPro-0.5.0.1\Julia-0
.5.0\lib\julia\sys.dll' --compile=yes --depwarn=yes --check-bounds=yes --code-co
verage=none --color=yes --compilecache=yes 'C:\Users\julia\AppData\Local\JuliaPr
o-0.5.0.1\pkgs-0.5.0.1\v0.5\FileIO\test\runtests.jl'`, ProcessExited(1)) [1]

================================================================================

INFO: Removing FactCheck v0.4.3
ERROR: FileIO had test errors
 in #test#61(::Bool, ::Function, ::Array{AbstractString,1}) at .\pkg\entry.jl:74
0
 in (::Base.Pkg.Entry.#kw##test)(::Array{Any,1}, ::Base.Pkg.Entry.#test, ::Array
{AbstractString,1}) at .\<missing>:0
 in (::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#test,Tuple{Array{Abstract
String,1}}})() at .\pkg\dir.jl:31
 in cd(::Base.Pkg.Dir.##2#3{Array{Any,1},Base.Pkg.Entry.#test,Tuple{Array{Abstra
ctString,1}}}, ::String) at .\file.jl:48
 in #cd#1(::Array{Any,1}, ::Function, ::Function, ::Array{AbstractString,1}, ::V
ararg{Array{AbstractString,1},N}) at .\pkg\dir.jl:31
 in (::Base.Pkg.Dir.#kw##cd)(::Array{Any,1}, ::Base.Pkg.Dir.#cd, ::Function, ::A
rray{AbstractString,1}, ::Vararg{Array{AbstractString,1},N}) at .\<missing>:0
 in #test#3(::Bool, ::Function, ::String, ::Vararg{String,N}) at .\pkg\pkg.jl:25
8
 in test(::String, ::Vararg{String,N}) at .\pkg\pkg.jl:258
```

### PR DESCRIPTION
**Pull request by SimonDanisch**
Fixes #82 and some formatting.
Also fixes some problems from the abspath branch.
@timholy, not really sure how that one survived the tests ;)

### CODE REVIEW

File: src/query.jl [hunk 1]
```diff
@@ -366,6 +366,7 @@ format inferred from the file's extension and/or magic bytes.
 """
 function query(filename::AbstractString)
     _, ext = splitext(filename)
+    filename = abspath(filename)
```
Comment (timholy) [inline src/query.jl]: I guess it's a question of when we want to "immortalize" the path. I was deliberate in my choices, thinking about this kind of chicanery:

```jl
tim@diva:~$ cd ~
tim@diva:~$ julia -q
julia> using FileIO, Colors

julia> f = query("photo.png")  # here I just want the format, but not the path
FileIO.File{FileIO.DataFormat{:PNG}}("photo.png")

julia> cd(mktempdir())

julia> open(f, "w") do io
           save(io, rand(RGB{U8}, 5, 5))
       end

<photo.png is in the tempdir, not my homedir>
```

I was thinking this is a good thing.

But perhaps it's reasonable to postulate that the moment of calling `query` constitutes the best time to immortalize. We should choose whatever is "least surprising" and then document the behavior. If you feel strongly that this version is less surprising, I say let's go with it.

Comment (SimonDanisch) [inline src/query.jl]: I haven't thought much about this admittedly, I just went with the version that results in the same behavior for all branches, which seemed like a good thing to me (and made the tests pass).
So I don't feel strongly about it and your argument makes sense. I guess you need this more, when working with Streams.

**Comment by aviks** [comment]
Is this ready to merge?

**Comment by musm** [comment]
Is this relevant anymore, `Pkg.test("FileIO")` spits the following on my windows machine
```julia
julia> Pkg.test("FileIO")
INFO: Computing test dependencies for FileIO...
INFO: No packages to install, update or remove
INFO: Testing FileIO
these tests will print warnings:
Error encountered while saving "test.not_installed".
Fatal error:
Library "NotInstalled" is not installed but is recommended as a library to load format: ".not_installed"
Should we install "NotInstalled" for you? (y/n):
Error encountered while saving "test.not_installed".
Fatal error:
Library "NotInstalled" is not installed but is recommended as a library to load format: ".not_installed"
Should we install "NotInstalled" for you? (y/n):
invalid is not a valid choice. Try typing y or n
Should we install "NotInstalled" for you? (y/n):
Error encountered while loading "Nullable{String}()".
Fatal error:
Error encountered while saving "Nullable{String}()".
Fatal error:
this test will print warnings:
Errors encountered while loading "test.multierr".
All errors:
   ErrorException("1")
   ErrorException("2")
Fatal error:
Test Summary: | Pass  Total
  FileIO      |  107    107
INFO: FileIO tests passed
INFO: No packages to install, update or remove
```

**Comment by SimonDanisch** [comment]
I think this can be closed! I just checked, the printed errors are all to be expected. We suppress STDERR to make it easier to recognize if the tests pass, but part of the errors weren't actually printed to STDERR. I opened: https://github.com/JuliaIO/FileIO.jl/pull/96

**Pull request closed without merging**